examples/stein_2d_toy.py

```diff
@@ -0,0 +1,235 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Example: 2d toy distribution with SteinVI
+===================================================
+In this example we infer a 2d star distribution with the different kernels available for SteinVI.
```

Comment:
I think we need to have a longer docstring to introduce SteinVI and its basic usage. By the way, could you also add a thumbnail for this example?

Reply:
That makes sense. I also started a [notebook version](https://github.com/aleatory-science/numpyro/blob/einstein_benchmark/notebooks/source/kernels.ipynb) of this example. Considering the intent of this example is usage, I think the notebook format is better. I'll open a separate PR with the notebook.